### PR TITLE
Bug 1150749 - Use bidi-isolate tags around track and artist names. r=djf

### PR DESCRIPTION
**Pull request by tedders1**

### CODE REVIEW

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=26a980ad7b1bd705aaf2c00f29667ead4001cb33

File: apps/system/js/media_playback.js [hunk 1]
```diff
@@ -155,19 +155,30 @@ MediaPlaybackWidget.prototype = {
...
-      track.push(artist);
-    }
-    track = track.join(' — '); // Using a &mdash; here.
-    if (track) {
+    var emDashNode = document.createTextNode(' — '); // Using a &mdash; here.
+
+    if (title || artist) {
       this.track.removeAttribute('data-l10n-id');
-      this.track.textContent = track;
+      this.track.innerHtml = '';
```
Comment (davidflanagan) [inline apps/system/js/media_playback.js]: The property name is innerHTML, not innerHtml.  But you should use textContent anyway. Anytime you use innerHTML, you're adding code that needs to be audited for script injection attacks. Clearly using the empty string here is safe, but it is even safer to use textContent.

File: apps/system/js/media_playback.js [hunk 2]
```diff
@@ -155,19 +155,30 @@ MediaPlaybackWidget.prototype = {
...
     var title = metadata.title.trim();
+    var titleNode = document.createTextNode(title || '');
+    var titleBdiNode = document.createElement('bdi');
+    titleBdiNode.appendChild(titleNode);
+
     var artist = metadata.artist.trim();
-    var track = [];
+    var artistNode = document.createTextNode(artist || '');
+    var artistBdiNode = document.createElement('bdi');
+    artistBdiNode.appendChild(artistNode);
```
Comment (davidflanagan) [inline apps/system/js/media_playback.js]: Please move this new code that creates the text nodes and bdi elements down below inside the `if (title)` and `if (artist)` blocks. As it stands, you're always creating these things even if they are not used.

File: apps/system/lockscreen/js/lockscreen_media_playback.js [hunk 3]
```diff
@@ -94,17 +94,33 @@ LockScreenMediaPlaybackWidget.prototype = {
...
+    var artist = metadata.artist.trim();
+    var artistNode = document.createTextNode(artist || '');
+    var artistBdiNode = document.createElement('bdi');
+    artistBdiNode.appendChild(artistNode);
+
+    var emDashNode = document.createTextNode(' — '); // Using a &mdash; here.
+
+    if (title || artist) {
+      this.track.removeAttribute('data-l10n-id');
+      this.track.innerHtml = '';
```
Comment (davidflanagan) [inline apps/system/lockscreen/js/lockscreen_media_playback.js]: Same comments here: innerHTML or textContent.

File: apps/system/lockscreen/js/lockscreen_media_playback.js [hunk 4]
```diff
@@ -94,17 +94,33 @@ LockScreenMediaPlaybackWidget.prototype = {
...
-      track.push(title);
-    }
-    if (artist) {
-      track.push(artist);
+    var artist = metadata.artist.trim();
+    var artistNode = document.createTextNode(artist || '');
+    var artistBdiNode = document.createElement('bdi');
+    artistBdiNode.appendChild(artistNode);
+
+    var emDashNode = document.createTextNode(' — '); // Using a &mdash; here.
```
Comment (davidflanagan) [inline apps/system/lockscreen/js/lockscreen_media_playback.js]: Same comments in this file. Please don't create the text nodes and bdi elements until you're in the if blocks below and are sure you'll actually need them.

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=44364944dcfa1ab8322241347a1cef14730b04e5

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=7d01e320912ef373778ff0d56fcd0bf4d97f4c38

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=2e815057b953511b049a3931749f00c8840f7f4a

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=fc9692b0c417918151852f8ab1c1223eca31e3e9

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=64a697b0eff8a4cdb8efd746bb23bc9277da969e

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=3f78524835d223023589b7378d1f121d8f585dd3

**Pull request merged**